### PR TITLE
don't autodeploy if preflights are failing

### PR DESCRIPTION
**Pull request by kcboyle**
#### What type of PR is this?
type::bug

#### What this PR does / why we need it:
This PR fixes a bug where auto-deploys would deploy updates to the application automatically even when the preflight checks were failing

#### Does this PR introduce a user-facing change?

```release-note
- fixes a bug with auto deploys where deployments would happen regardless of preflight success or failure. Auto-deploys will now only deploy if the preflights succeed
```

#### Does this PR require documentation?
NONE

### CODE REVIEW

**Comment by kcboyle** [comment]
Screen Shot 2022-03-11 at 1 57 18 PM
Screenshot of log message when the preflights fail

**Review by DanStough — APPROVED** [review]

File: pkg/updatechecker/updatechecker.go [hunk 1]
```diff
@@ -513,13 +516,49 @@ func autoDeploy(opts CheckForUpdatesOpts, clusterID string, autoDeploy apptypes.
...
+	}
+
+	var parsedResults preflight.UploadPreflightResults
+	if err = json.Unmarshal([]byte(preflightResult.Result), &parsedResults); err != nil {
+		return errors.Wrap(err, "failed to parse preflight results")
+	}
+
+	if parsedResults.Errors != nil {
+		return errors.Wrap(errors.New(preflightResult.Result), "preflight errors")
+	}
```
Comment (pavansokkenagaraj) [inline pkg/updatechecker/updatechecker.go:557]: preflight result doesn't account for failed analyzer

preflightresult.results[].isFail=true, the deployments should be blocked.

+ when there is no preflight spec in the app, we may need to test the edge case

**Review by pavansokkenagaraj — CHANGES_REQUESTED** [review]

**Comment by kcboyle** [comment]
messed up the rebase on this one. opening up a [new PR](https://github.com/replicatedhq/kots/pull/2628) that is fixed

File: pkg/updatechecker/updatechecker.go [hunk 2]
```diff
@@ -513,13 +517,53 @@ func autoDeploy(opts CheckForUpdatesOpts, clusterID string, autoDeploy apptypes.
...
+	if err != nil {
+		return errors.Wrap(err, "failed to poll for preflights results")
+	}
+	// refetch latest results
+	preflightResult, err := store.GetPreflightResults(appID, sequence)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch preflight results")
+	}
+
+	if preflightResult == nil || len(preflightResult.Result) == 0 {
```
Comment (sgalsaleh) [inline pkg/updatechecker/updatechecker.go:551]: this should not fail, some apps don't have preflights
